packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,301 @@
+    if (props.additionalBehaviors) {
+      Object.entries(props.additionalBehaviors).forEach(([pathPattern, behaviorOptions]) => {
+        this.addBehavior(pathPattern, behaviorOptions);
+      });
+    }
+
+    this.certificate = props.certificate;
+    this.errorConfigurations = props.errorConfigurations ?? [];
+
+    const distribution = new CfnDistribution(this, 'CFDistribution', { distributionConfig: {
+      enabled: true,
+      origins: Lazy.anyValue({ produce: () => this.renderOrigins() }),
+      defaultCacheBehavior: this.defaultBehavior._renderBehavior(),
+      cacheBehaviors: Lazy.anyValue({ produce: () => this.renderCacheBehaviors() }),
+      viewerCertificate: this.certificate ? { acmCertificateArn: this.certificate.certificateArn } : undefined,
+      customErrorResponses: this.renderCustomErrorResponses(),
+      priceClass: props.priceClass ?? undefined,
+    } });
+
+    this.domainName = distribution.attrDomainName;
+    this.distributionId = distribution.ref;
+  }
+
+  /**
+   * Adds a new behavior to this distribution for the given pathPattern.
+   *
+   * @param pathPattern the path pattern (e.g., 'images/*') that specifies which requests to apply the behavior to.
+   * @param behaviorOptions the options for the behavior at this path.
+   */
+  public addBehavior(pathPattern: string, behaviorOptions: BehaviorOptions) {
+    if (pathPattern === '*') {
+      throw new Error('Only the default behavior can have a path pattern of \'*\'');
+    }
+    this.additionalBehaviors.push(new CacheBehavior({ pathPattern, ...behaviorOptions }));
+  }
+
+  private renderOrigins(): CfnDistribution.OriginProperty[] {
+    const origins = new Set<Origin>();
+    origins.add(this.defaultBehavior.origin);
+    this.additionalBehaviors.forEach(behavior => origins.add(behavior.origin));
+
+    const renderedOrigins: CfnDistribution.OriginProperty[] = [];
+    origins.forEach(origin => renderedOrigins.push(origin._renderOrigin()));
+    return renderedOrigins;
+  }
+
+  private renderCacheBehaviors(): CfnDistribution.CacheBehaviorProperty[] | undefined {
+    if (this.additionalBehaviors.length === 0) { return undefined; }
+    return this.additionalBehaviors.map(behavior => behavior._renderBehavior());
+  }
+
+  private renderCustomErrorResponses(): CfnDistribution.CustomErrorResponseProperty[] | undefined {
+    if (this.errorConfigurations.length === 0) { return undefined; }
+    function validateCustomErrorResponse(errorResponse: CfnDistribution.CustomErrorResponseProperty) {
+      if (errorResponse.responsePagePath && !errorResponse.responseCode) {
+        throw new Error('\'responseCode\' must be provided if \'responsePagePath\' is defined');
+      }
+      if (!errorResponse.responseCode && !errorResponse.errorCachingMinTtl) {
+        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
```

Comment:
Its ok to use external double-quotes in case you need an internal single quoted string. Linter won't yell, and it's easier to read.

Reply:
:) I've seen both used within the code base -- and both stated as preferable by different team members. Happy to switch to double-quoting though.